### ISSUE TITLE
Unable to fold with an accumulator of a type with a wildcard parameter.

### ISSUE THREAD

**Issue by noresttherein** [scala/bug#12749]
## Reproduction steps

Scala version: 2.13.10

```scala
	type ~[A, B]

	class Test[V] {
		def ~[L](t :Test[L]) :Test[L ~ L] = ???
	}

	def tests :Seq[Test[_]] = ???

	((??? :Test[_]) /: tests) { (acc, test) => acc ~ test }
```

## Problem
```
type mismatch;
 found   : Playground.Test[_$1 ~ _$1] where type _$1
 required: Playground.Test[_$2] where type _$2
	((??? :Test[_]) /: tests) { (acc, test) => acc ~ test }
```
I know existentials are a bit of a mess with ambiguity regarding quantifiers, but this gets a bit annoying. Works in Scala 3, FWIW.

**Comment by joroKr21** [comment]
Isn't that the same as `List(1,2,3).foldRight(Nil)(_ :: _)` not working?
I couldn't get it to work in Scala 3 either: https://scastie.scala-lang.org/6cUR8UbhRf6upfLwUeIYYg

**Comment by joroKr21** [comment]
Yes I think it's the opposite case so I agree that it should be working

**Comment by noresttherein** [comment]
As far as I understand/remember, the first case doesn't work because the typer infers `List[Nothing]` as the accumulator type,
and `List[Nothing]::[Int]` is not a `List[Nothing]`, so you need
```Scala
List(1, 2, 3).foldRight(Nil :List[Int])(::)
```
Unless something has changed, or I do not understand, it is a completely different case, because in my example no amount of extra type declarations and hinting could make the compiler treat two `Test[_]` as the same type. Yes, I know that wildcards sometimes are meant to mean 'Test[T] for some type `T`' and sometimes 'Test[T] for any `T`', which seems like a hopeless case, and the fact that Scala 3 seems to inherit this duality is not encouraging.
